src/NUnitFramework/framework/Internal/TestExecutionContext.cs

```diff
@@ -484,9 +484,19 @@ private TestExecutionContext CreateIsolatedContext()
             return context;
         }
 
-#endregion
+        /// <summary>
+        /// Sends a message from test to listeners. This message is not kind of test output and doesn't go to test result.
+        /// </summary>
+        /// <param name="destination">A name recognized by the intended listeners.</param>
+        /// <param name="message">A message to be sent</param>
+        public void SendMessage(string destination, string message)
+        {
+            Listener?.SendMessage(new TestMessage(destination, message, CurrentContext.CurrentTest.Id));
```

Comment:
This is an instance method - as it should be. So the user already called the static `CurrentContext` to get the context that is to be used. In that scenario, your use of it is merely redundant.

But consider a unit test, where we create our own `TestExecutionContext` so as not to interfere with other tests. In that case, your code is not redundant but causes an error because the __actual__ current context will be used instead of the one we created for testing!